internal/irc/handler.go

```diff
@@ -220,6 +220,7 @@ func (h *Handler) Run() error {
 				h.log.Debug().Msgf("connect attempt %d", connectAttempts)
 
 				if err := h.client.Connect(); err != nil {
+					h.log.Debug().Msgf("encountered connection error: %+v", err)
```

Comment:
This should probably be an error log instead of debug.

```diff
-h.log.Debug().Msgf("encountered connection error: %+v", err)
+h.log.Error().Err(err).Msgf("client encountered connection error")
```